### ISSUE TITLE
[Spree Upgrade] Google Maps "NoApiKeys" warning in feature specs

### ISSUE THREAD

**Issue by HugsDaniel** [openfoodfoundation/openfoodnetwork#3096]
## Description

When running feature specs in 2-0-stable, I get this error : 
`Google Maps JavaScript API warning: NoApiKeys https://developers.google.com/maps/documentation/javascript/error-messages#no-api-keys`

I last got it when running `spec/features/consumer/shopping/checkout_spec.rb`, but it's the same for several feature specs. 

It doesn't make the specs fail, but it takes a lot of time by retrying 3 or 4 times to connect to Google API.

As @sauloperez and @mkllnk suggested, I open this issue so that we can investigate how not to depend on internet connection for specs.

**Comment by sauloperez** [comment]
IMO this is a perfect fit for https://github.com/bblimke/webmock

**Comment by mkllnk** [comment]
This is not a warning anymore: https://semaphoreci.com/openfoodfoundation/openfoodnetwork-2/branches/master/builds/347

Our build is broken. We need to fix this as soon as possible.

**Comment by luisramos0** [comment]
broken build, we stop the "production line" https://en.wikipedia.org/wiki/Andon_(manufacturing) :-)

**Comment by sauloperez** [comment]
I gave it a try and Webmock can't help us in this case since it's not a regular HTTP request made through a Ruby HTTP client but a <script> tag.

I believe the culprit is https://github.com/openfoodfoundation/openfoodnetwork/blob/66585e93a85f1d557a54345d52af1bac822f3a1f/app/views/layouts/darkswarm.html.haml#L19.

I also tried adding a guard clause with `Rails.env.production?` but darkswarm depends on the `google` object. See below:

```
  1) Shops searches by URL
     Failure/Error: visit shops_path(anchor: "/?query=xyzzy")

     Capybara::Poltergeist::JavascriptError:
       One or more errors were raised in the Javascript code on the page. If you don't care about these errors, you can ignore them by setting js_errors: false in your Poltergeist configuration (see documentation for details).

       ReferenceError: Can't find variable: google
       ReferenceError: Can't find variable: google
           at http://127.0.0.1:50035/assets/darkswarm/all.js:67070 in global code
       ReferenceError: Can't find variable: Darkswarm
       ReferenceError: Can't find variable: Darkswarm
           at http://127.0.0.1:50035/assets/web/all.js:2
       ReferenceError: Can't find variable: I18n
       ReferenceError: Can't find variable: I18n
           at http://127.0.0.1:50035/shops#/?query=xyzzy:19 in global code
       Error: [$injector:nomod] Module 'Darkswarm' is not available! You either misspelled the module name or forgot to load it. If registering a module ensure that you specify the dependencies as the second argument.
       http://errors.angularjs.org/1.5.5/$injector/nomod?p0=Darkswarm
```

**Comment by luisramos0** [comment]
I'd go for:
1 - use Rails.env.production? and mock the google object with whatever is essential for the website to work (one option is to make a real call to the api and store the result somewhere in test support)
or
2 - add an API key to the build (I dont think it will be expensive)

**Comment by kristinalim** [comment]
There's [`PuffingBillly`](https://github.com/oesmith/puffing-billy) which sets up a proxy for the browser and allows recording and stubbing browser requests. It should work like webmock and [`VCR`](https://github.com/vcr/vcr) for the browser. I think we should explore this option.

But I have to note that I had tried to get this to work on Ruby 1.9.3 / Rails 3.2.x / selenium-webdriver 2.53.4 / eventmachine_httpserver 0.2.1 before and wasn't successful, so I'd be cautious about spending too much time on this if this doesn't seem to work with our current library versions.

**Comment by luisramos0** [comment]
yes, also because these two specs are not the most important (search by url). #3168 has a green build and awaits reviews ;-)

**Comment by filipefurtad0** [comment]
Hey @RachL ,
Yes - the build was fixed by skipping these specs, but the issue persists. I guess this would be a case where VCR or puffing billy would be appropriate as  Kristina suggested here https://github.com/openfoodfoundation/openfoodnetwork/issues/3096#issuecomment-444052352.

Keeping it open in the welcome new QAs board seems right :+1:
